### PR TITLE
#181414718 (Feature): login with email & password

### PR DESCRIPTION
**Pull request by Robertishimwe**
### What does this PR do?

Users should be able to Login to Barefoot Nomad

### Description of Task to be completed?

**GIVEN** user successfully logs in.
**WHEN** user passes appropriate login parameters 
**THEN** user should be able to login.

### How should this be manually tested?
Get started By Running commands bellow:
  - clone repo
  - switch to [ft-login-181414718](https://github.com/atlp-rwanda/code-ninjas-fe/tree/ft-login-181414718) 
  - branch
  - Run `yarn install`: to install all dependencies
  - Run `yarn dev`: to start the application
  - Open your browser & visit  `/controlpannel` route.  you can not access is, this is because it is protected. only logged in users have access to this.
  - Now visit `/login` route & and enter your credentials(verified account). now you will be redirected to `/controlpannel` you was not able to access before.

`

Routes:

`

[/login](https)
login PAGE  `Register`

[/controlpannel](https)
login PAGE  `Register`

`

### What are the relevant pivotal tracker stories?
[#181414716](https://www.pivotaltracker.com/story/show/181414716)
### Questions:

### CODE REVIEW

File: src/components/login/loginForm.jsx [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+  CardActions,
+  TextField,
+  Typography,
+  Button,
+} from '@material-ui/core';
+import { thisUser, logginUser } from '../../features/auth/loginSlice';
+
+import google from '../../asset/icons8-google.svg';
+import facebook from '../../asset/icons8-facebook.svg';
+// import validationsForm from '../../validation/registrationValidation';
```
Comment (ericus123) [inline src/components/login/loginForm.jsx]: remove this

File: src/styles/footer.scss [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+.footer {
+    background-color: $primaryColor;
+    color: $secondaryColor;
+    padding: 5px 0px;
+    text-align: center;
+    font-family: $text, sans-serif;
+    font-weight: bolder;
+    // z-index: 9;
+    // position: fixed;
+    // bottom: 0;
```
Comment (ericus123) [inline src/styles/footer.scss]: remove this

**Review by ericus123 — CHANGES_REQUESTED** [review]

**Review by niyojuste — COMMENTED** [review]
- Once the registration is available, you can maybe add a link to **sign up** in case the user doesn't have an account yet
- Also, can you add an option for users to view their password as they input it

**Comment by Robertishimwe** [comment]
> * Once the registration is available, you can maybe add a link to **sign up** in case the user doesn't have an account yet
> * Also, can you add an option for users to view their password as they input it

**Review by Procech20 — DISMISSED** [review]

**Comment by Robertishimwe** [comment]
> > * Once the registration is available, you can maybe add a link to **sign up** in case the user doesn't have an account yet >i wii add this after rebasing from registration
> > * Also, can you add an option for users to view their password as they input it >Done

File: src/components/login/loginForm.jsx [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+              error={touched.password && Boolean(errors.password)}
+              margin="dense"
+              variant="outlined"
+              fullWidth
+            />
+            <FormControlLabel
+              control={<Checkbox defaultChecked />}
+              label="Show password"
+              onClick={showPassword}
+            />
```
Comment (niyojuste) [inline src/components/login/loginForm.jsx]: Nicely done! 
One small detail though, the logic is inverted. You can either start with:
- `Checkbox defaultChecked` in **FormControlLabel** with `type="text"` in **TextField** _or_
- `Checkbox` only in **FormControlLabel** with `type="password"` in **TextField**

**Review by niyojuste — DISMISSED** [review]

**Review by itisWasp — DISMISSED** [review]

**Review by Procech20 — DISMISSED** [review]

File: src/components/login/socialAuth.jsx [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  window.location.href.replace(/[?&]+([^=&]+)=([^&]*)/gi, (m, key, value) => {
+    vars[key] = value;
+  });
+
+  const { token } = vars;
+  const decoded = jwtDecode(token);
+  const user = { user: decoded.user, token };
+  dispatch(logginUser(user));
+  dispatch(loginMode());
+  // eslint-disable-next-line no-unused-expressions
```
Comment (ericus123) [inline src/components/login/socialAuth.jsx]: why are you disabling this rule???

**Review by ericus123 — CHANGES_REQUESTED** [review]
@Robertishimwe  Work on these changes as well as make sure you are really testing.
You should test the form input fields validations too. **ex:** expect an error if a particular input field is not field or is invalid

File: src/components/login/loginForm.jsx [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+    window.open('https://barefoot.onrender.com/api/auth/facebook', '_blank');
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    axios
+      .post('https://barefoot-nomad-v2.herokuapp.com/api/auth/login', {
+        email,
+        password,
+      })
```
Comment (ericus123) [inline src/components/login/loginForm.jsx]: configure Axios globally. with some basic headers that are being used ex: token.

File: src/redux/store.js [hunk 6]
```diff
@@ -1,32 +1,32 @@
...
-  PERSIST,
-  PURGE,
-  REGISTER,
+    persistReducer,
+    FLUSH,
+    REHYDRATE,
+    PAUSE,
+    PERSIST,
+    PURGE,
+    REGISTER,
```
Comment (ericus123) [inline src/redux/store.js]: why have unnecessary indentation changes ??? 
Are you using prettier configuration as configured?

**Review by ericus123 — CHANGES_REQUESTED** [review]

File: src/components/login/loginForm.jsx [hunk 7]
```diff
@@ -0,0 +1,224 @@
...
+  };
+
+  const facebookFun = () => {
+    window.open('https://barefoot.onrender.com/api/auth/facebook', '_blank');
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    axios
+      .post('https://barefoot-nomad-v2.herokuapp.com/api/auth/login', {
```
Comment (ericus123) [inline src/components/login/loginForm.jsx]: why is this hardcoded ????

File: src/components/login/loginForm.jsx [hunk 8]
```diff
@@ -0,0 +1,224 @@
...
+    const x = document.getElementById('password');
+    if (x.type === 'password') {
+      x.type = 'text';
+    } else {
+      x.type = 'password';
+    }
+  };
+
+  const googleFun = () => {
+    window.open('https://barefoot.onrender.com/api/auth/google', '_blank');
```
Comment (ericus123) [inline src/components/login/loginForm.jsx]: this shouldn't be hardcoded

File: src/components/login/loginForm.jsx [hunk 9]
```diff
@@ -0,0 +1,224 @@
...
+        dispatch(loginMode());
+      })
+      .catch((err) => {
+        const { error } = err.response.data;
+        const displayError = document.getElementsByClassName('displayer');
+        displayError[0].innerText = error;
+      });
+  };
+
+  // eslint-disable-next-line no-unused-expressions
```
Comment (ericus123) [inline src/components/login/loginForm.jsx]: don't disable eslint rules. remove this and wherever you are disabling eslint rules

**Review by ericus123 — APPROVED** [review]

**Pull request merged**